src/pages/index.js

```diff
@@ -79,42 +148,133 @@ const addCardPopUp = new PopUpWithForm(cardPopUpSelector, handleAddCardSubmit);
 /** FUNCTIONS **/
 
 /* CARD */
-// Re - why we keep createCard in index.js again? Why not in Card.js?
 function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleTrashClick,
+    handleLikeClick
+  );
   return card.generateCard();
 }
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit(data) {
-  user.setUserInfo(data.title, data.description);
+  api
+    .updateUserProfile({ name: data.title, about: data.description })
+    .then((res) => {
+      user.setUserInfo(res.name, res.about);
+      profileEditPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Profile Patch ${err}`);
+    })
+    .finally(() => {
+      profileEditPopUp.setSaving(false);
+    });
+
   profileEditPopUp.close();
```

Comment:
`  profileEditPopUp.close();
`

It's not needed here. You have it in `then`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**